src/whylogs/core/datasetprofile.py

```diff
@@ -796,3 +799,34 @@ def array_profile(
     prof = DatasetProfile(name, timestamp)
     prof.track_array(x, columns)
     return prof
+
+
+def _create_update_summary_dictionary(number_summary: NumberSummary, **kwargs):
+    """
+    Wrapper method for summary constraints update object creation
+
+    Parameters
+    ----------
+    number_summary : NumberSummary
+        Summary object generated from NumberTracker
+        Used to unpack the metrics as separate items in the dictionary
+    kwargs : Summary objects or datasketches objects
+        Used to update specific constraints that need additional calculations
+
+    Returns
+    -------
+    Anonymous object containing all of the metrics as fields with their coresponding values
+    """
+
+    update_dict = {}
+
+    update_dict.update(
+        {
+            field_name: getattr(number_summary, field_name)
+            for field_name in dir(number_summary)
+            if str.islower(field_name) and not str.startswith(field_name, "_") and not callable(getattr(number_summary, field_name))
+        }
+    )
+    update_dict.update(kwargs)
+
+    return type("Object", (), update_dict)
```

Comment:
This is better factored into a utility function. I think we should revisit this and consider replacing with a facade class, we can also update the consuming code to not require this shape to make it easier to compose the column profile and summary fields we need access to.